cmd/snap-bootstrap/triggerwatch/triggerwatch.go

```diff
@@ -0,0 +1,88 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package triggerwatch
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/snapcore/snapd/logger"
+)
+
+type triggerProvider interface {
+	FindMatchingDevices(filter triggerEventFilter) ([]triggerDevice, error)
+}
+
+type triggerDevice interface {
+	WaitForTrigger(chan keyEvent)
+	String() string
+	Close()
+}
+
+var (
+	// trigger mechanism
+	trigger triggerProvider
+
+	// wait for '1' to be pressed
+	triggerFilter = triggerEventFilter{Key: "KEY_1"}
+
+	ErrTriggerNotDetected = errors.New("trigger not detected")
+)
+
+// Wait waits for a trigger on the available trigger devices for a given amount
+// of time. Returns nil if one was detected, ErrTriggerNotDetected if timeout
+// was hit, or other non-nil error.
+func Wait(timeout time.Duration) error {
+	if trigger == nil {
+		logger.Panicf("trigger is unset")
+	}
+
+	devices, err := trigger.FindMatchingDevices(triggerFilter)
+	if err != nil {
+		return fmt.Errorf("cannot list trigger devices: %v", err)
+	}
+	if devices == nil {
+		return fmt.Errorf("cannot find matching devices")
+	}
+
+	logger.Noticef("waiting for trigger key: %v", triggerFilter.Key)
+
+	// wait for a couple of second for the key
+	detectKeyCh := make(chan keyEvent, len(devices))
+
+	for _, kbd := range devices {
+		go kbd.WaitForTrigger(detectKeyCh)
```

Comment:
I would prefer this to be `dev` instead of `kbd` like:

```go
for _, dev := range devices {
    go dev.WaitForTrigger(detectKeyCh)
    defer dev.Close()
}
```

This is a bit more consistent and reduces the likelihood that when we do support non-keyboard devices for input that we have old variable names like this to confuse folks.